generated/src/aws-cpp-sdk-s3-crt/source/S3CrtErrorMarshaller.cpp

```diff
@@ -96,4 +101,43 @@ Aws::String S3CrtErrorMarshaller::ExtractEndpoint(const AWSError<CoreErrors>& er
   }
 
   return {};
+}
+AWSError<Aws::Client::CoreErrors> S3CrtErrorMarshaller::Marshall(const Aws::Http::HttpResponse& httpResponse) const
+{
+  //if response code not ok, use error marshaller
+  if(httpResponse.GetResponseCode() != HttpResponseCode::OK)
+  {
+    return XmlErrorMarshaller::Marshall(httpResponse);
+  }
+
+  Aws::String message{"Error in body of the response"};
+  //extract error message and code in the body
+  auto& body = httpResponse.GetResponseBody();
+  auto readPointer = body.tellg();
+  XmlDocument doc = XmlDocument::CreateFromXmlStream(body);
+  Aws::String bodyError;
+
+  if (doc.WasParseSuccessful() &&
+      !doc.GetRootElement().IsNull() && 
+      doc.GetRootElement().GetName() == Aws::String("Error")) 
+  {        
+      auto messageNode = doc.GetRootElement().FirstChild("Message") ;
+      if(!messageNode.IsNull())
+      {
+          message = messageNode.GetText();
+      }
+      auto codeNode = doc.GetRootElement().FirstChild("Code") ;
+      if(!codeNode.IsNull())
+      {
+          bodyError = codeNode.GetText();
+      }
+  }
+  body.seekg(readPointer);
```

Comment:
nit: I think it would be better to rewind the stream right after parsing (anyway parsing is done by copying the stream into another string).

Reply:
ack, in this case it won't matter since the steam was already parsed and string copied over already by this point. but I can move it to the end of the function.